ci-operator/step-registry/cucushift/installer/check/gcp/nested-virtualization/cucushift-installer-check-gcp-nested-virtualization-commands.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+# to be able to SSH.
+if ! whoami &> /dev/null; then
+    if [[ -w /etc/passwd ]]; then
+        echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    else
+        echo "/etc/passwd is not writeable, and user matching this uid is not found."
+        exit 1
+    fi
+fi
+
+if [ ! -f "${SHARED_DIR}/bastion_public_address" ] || [ ! -f "${SHARED_DIR}/bastion_ssh_user" ]; then
+    echo "ERROR: Failed to get bastion host info, abort. " && exit 1
+fi
+
+function run_command() {
+    local cmd="$1"
+    echo "Running Command: ${cmd}"
+    eval "${cmd}"
+}
+
+function run_ssh_cmd() {
+    local sshkey=$1
+    local user=$2
+    local host=$3
+    local remote_cmd=$4
+
+    options=" -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null "
+    cmd="ssh ${options} -i \"${sshkey}\" ${user}@${host} \"${remote_cmd}\" 1>stdout 2>stderr"
+    run_command "$cmd" || return 2
+    return 0
+}
+
+function run_scp_to_remote() {
+    local sshkey=$1
+    local user=$2
+    local host=$3
+    local src=$4
+    local dest=$5
+
+    options=" -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null "
+    cmd="scp ${options} -i \"${sshkey}\" ${src} ${user}@${host}:${dest}"
+    run_command "$cmd" || return 2
+    return 0
+}
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+ssh_key_file_name="ssh-privatekey"
+ssh_key=${CLUSTER_PROFILE_DIR}/${ssh_key_file_name}
+bastion_dns=$(head -n 1 "${SHARED_DIR}/bastion_public_address")
+bastion_user=$(head -n 1 "${SHARED_DIR}/bastion_ssh_user")
+
+working_dir=$(mktemp -d)
+pushd "${working_dir}"
+
+# SCP ssh key file to the bastion host
```

Comment:
I think we do not need to copy the ssh key to bastion host, but use `ProxyCommand` to run remote command.

```
function _ssh() {
    echo "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no ${PROXYCOMMAND} $@" | sh -
}
PROXYCOMMAND="-o ProxyCommand='ssh -o StrictHostKeyChecking=no -W %h:%p -i '${ssh_key}' ${bastion_user}@${bastion_dns}'"
_ssh -i '${ssh_key}' ${bastion_user}@${bastion_dns} "grep -cw vmx /proc/cpuinfo"
```